src/resources/Groups/GroupsInterfaces.ts

```diff
@@ -0,0 +1,71 @@
+export interface GroupModel {
+    id: string;
+    displayName: string;
+    invites?: InviteModel[];
+    members?: MemberModel[];
+    privileges?: PrivilegeModel[];
+    realms?: RealmModel[];
+    deletable?: boolean;
+    resourceId?: string;
+    groupsThatCanEdit?: IdAndDisplayNameModel[];
+    apiKeysThatCanEdit?: IdAndDisplayNameModel[];
+}
+
+export interface IdAndDisplayNameModel {
+    id: string;
+    displayName?: string;
+}
+
+export interface InviteModel {
+    id: string;
+    email: string;
+    username?: string;
+    providerUsername?: string;
+    displayName?: string;
+    expirationDate?: number;
+    invitedDate?: number;
+    provider?: AuthProvider;
+}
+
+export interface MemberModel {
+    id: string;
+    username: string;
+    email?: string;
+    displayName?: string;
+    lastLogin?: number;
+    provider?: AuthProvider;
+    providerUsername?: string;
+}
+
+export interface PrivilegeModel {
+    owner: string;
+    targetDomain: string;
+    targetId: string;
```

Comment:
`targetId` defaults to `*` btw

Reply:
so it its optional?

Reply:
technically yes, however if you set it to `"targetId: null` it will throw so handle it how you want